### DISCUSSION TITLE
Insights: Add related columns to selection list for axes

### DISCUSSION THREAD

**Discussion by DustinBracy** [directus/directus#18833] [Feature Requests]
### Summary

First I want to say that I love the new insight modules!

However, creating a new plot only shows the ID for relationship fields in a collection. This is a problem for fairly normalized databases, where many tables have foreign key relationships (all of them, in fact, if you want to use the 'user_created' field). The labels end up being represented entirely of UUIDs, which are not very user friendly.

The ability to select related column names, or even respect the default display template configured on the collection relationship field would go a very long way into making plots more usable.  

### Motivation

Bar and Line Charts are almost unusable on collections using relationships.  The below chart doesn't provide any business insight without manually looking up each UUID. 

Showing the user names (first/last), instead of the user ID here would be infinitely more helpful to anyone looking at the chart. 

![directus plot](https://github.com/directus/directus/assets/34081222/a7f5a2b6-1907-4957-a027-ac9dcf9b67f4)

### Detailed Design

Add related fields with dropdown in the X/Y axis selection boxes, similar to how we have in the column add section of the collection table interface.

Current Panel X/Y Axis Selection:
![image](https://github.com/directus/directus/assets/34081222/def42bbe-a1b6-482d-be4e-e68ef273398f)

Desired Panel X/Y Axis Selection:
![related fields](https://github.com/directus/directus/assets/34081222/1fa1124d-27d9-474f-8d54-4cf9ca7f8734)

### Must Have:
- The ability to select other columns from the foreign table in as axis labels.

### Could Have:
- The ability to modify the display template of the plot labels for each axis.
- The ability to control sort by fields other than the display axis.

### Drawbacks

- It requires additional code to pull in related fields.
- Possibly impacts plot performance with deeply nested relationships.

### Alternatives

Create plots in an outside application.

### Adoption Strategy

No breaking changes, this would simply be an additional swatch of fields to choose from for label purposes.

**Comment by emahuni** [comment]
When I saw this limitation I thought I was missing something somehow. But it seems like it's a real thing.

**Reply by ImAnAutie** [reply]
Same here, it feels like im missing something obvious!

**Comment by LevelUpGit** [comment]
+1 It's really required, data has no sense without it.

**Comment by gloribus** [comment]
++++1

It would also be cool if it would support displaying text not values if the field is a dropdown

**Reply by Seinnd** [reply]
👍

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by fredjoz** [reply]
++++1
this function is really missing

**Reply by trinly01** [reply]
@rijkvanzanten  Do you happen to have any updates on this one?

**Comment by photographycoursenet** [comment]
+1 for this

**Comment by devMls** [comment]
+1 for this

**Comment by xhulianobushi** [comment]
+1 for this

**Comment by floux-digital** [comment]
Some news about it?

+1,000 for this.... and, I miss the possibility of adding a global date filter (from / to) to control the dashboard more easily.

**Comment by gla23** [comment]
Yep agree this feels pretty essential when first trying out the graphs!

As well as letting users change the field that goes in the column, perhaps it should automatically choose a sensible default. For example when first creating the graph in Insights you'd want it to be relevant. Perhaps it should just use the Display Template for the relevant collection (like M2O and M2M fields do when their "Interface Display Template" is blank) or even matching for a field ending with `_name`.

**Comment by izoukhai** [comment]
+1 also

This is very very crucial

**Comment by jacobwise** [comment]
+1

**Comment by uzarsalan** [comment]
+1

**Comment by devoluxbv** [comment]
+1

**Comment by HIDAN122** [comment]
+1

**Comment by satwal** [comment]
+1

**Comment by NilsBaumgartner1994** [comment]
+1

**Comment by emwadde** [comment]
+1

**Comment by NilsBaumgartner1994** [comment]
Synonyms for specific values would be nice too !
Example you have the field "like" which has values "true" / "false"
You would like to have a translation "Likes" / "Dislikes"

**Comment by NilsBaumgartner1994** [comment]
Any updates on this feature?

**Comment by alepietrobon** [comment]
Not seeing a lot of movement here, so I'm adding another +1. for good fun.

**Comment by b45t3rr** [comment]
I see this as absolutely necessary, especially since the idea is to illustrate data that are almost always related.

**Comment by Flamintus** [comment]
+1

**Comment by devoluxbv** [comment]
What was point to add data dashboard if it is unusable without this feature

**Comment by gloribus** [comment]
Unfortunately, it feels like we're not being heard. This request has 89 upvotes, many comments, and it's been over two years.  
I genuinely love Directus and your whole team - I still remember how excited I was years ago when I first discovered the project. It truly felt like something special.

That's why it's so disheartening to see the recent direction. Instead of focusing on long-requested and truly helpful features (at the very least, just approve this one), the roadmap seems to be drifting toward things that feel disconnected from what the community is asking for. A quick look at recent updates makes that quite clear.

It's sad because there's still no real alternative to how amazing this product is. I truly hope to see it continue to grow and evolve.

**Reply by rijkvanzanten** [reply]
Our roadmap has indeed focussed a bit more on CMS and measured customer-value feature requests than exclusively github discussion upvotes, but that doesn't negate your feelings. I agree the transparency has taken a bit of a hit as we find better tooling to keep up with the community growth. GitHub's lack of tooling to keep up with changes in upvotes and comments is starting to become a real problem on my end.. 😬 I've been experimenting with [productlane](https://roadmap.directus.io), but where we'd gain a lot of help in processing feedback we'd lose some of the public visibility of existing threads. Hopefully that can be resolved through the public forum we just launched on https://community.directus.io/ though 🤞🏻

**Reply by gloribus** [reply]
@rijkvanzanten 
Thanks for the honest reply - really appreciate it!

Just to be sure: should we now discuss new features mainly on the forum (community.directus.io)?

I do hope the community can adapt - many contribute through PRs here on GitHub and want to stay involved in the open-source side of the project

**Reply by rijkvanzanten** [reply]
Oh I'm not ready to flip that switch yet! For now this is still the place to go 🙂 GitHub will stick around for bugs and PRs (and the source code obviously), but I do need to find a better solution for discussing/requesting new features.

We just switched from our Discord server to [Discourse](https://community.directus.io) as primary place for people asking help, which already has been a _tremendous_ improvement in making sure people get answers to the questions they have and to make sure people googling/chatgpting their questions find those answers as well. Up next is solving feature requests 🙂 

PS thanks to the person reading along for immediately trying my productlane experiment haha ✊🏻

**Reply by alepietrobon** [reply]
@rijkvanzanten , I appreciate the follow up and the clarity on process for requests and follow up.

However, you've not yet addressed the actual timeline for this feature. Should we take it that it's not a priority for your roadmap for the foreseeable future? 

I understand if that's the case, just wanted an update so we can work around it.

**Reply by rijkvanzanten** [reply]
This particular one is not currently top of our todo list no, but I'll happily take a PR if somebody wants to take a crack at it before we get to it 🙂

**Reply by ciremusyoka** [reply]
@rijkvanzanten I’d be happy to take a crack at implementing this feature. Looking forward to contributing and putting together a PR.

**Reply by rijkvanzanten** [reply]
Go for it @ciremusyoka! 🙌

**Reply by ALXRITR** [reply]
any news here?
so dumb, to have a metrics-dashboard but not being able to see the employee-names but some random-ass numbers ...

**Reply by JamesW1** [reply]
Hey! Yes, relational field support on the bar chart was merged here (available in upcoming 11.15.x release)
https://github.com/directus/directus/pull/26489

It felt like the x-axis was the primary benefit, given Y-axis in bar/line charts is almost always used for quantitative/numerical aggregations. That being said, if you have a strong example for adding the same to the y-axis, it should be straightforward to add.
